### ISSUE TITLE
Multithreading mode seed does not work as intended

### ISSUE THREAD

**Issue by lobis** [rest-for-physics/restG4#100]
https://rest-forum.unizar.es/t/multithreading-same-seed-same-number-of-threads-gives-different-simulations/578/4

**Comment by jgalan** [comment]
I don't believe it is related since I imagine Geant4 uses its own random generator. But just in case, I got some issues with Random generators here: https://root-forum.cern.ch/t/trandom3-problem-setting-seed/53850

BTW I need your help @lobis on pre-commit configuration here https://github.com/rest-for-physics/axionlib/pull/52
